### PR TITLE
Fix a few issues.

### PR DESCRIPTION
**Pull request by ildyria**

### CODE REVIEW

**Comment by ildyria** [comment]
This PR will also allow `php artisan lychee:Diagnostics` to work even without database connection.

File: app/Configs.php [hunk 1]
```diff
@@ -259,13 +259,19 @@ public static function hasExiftool()
...
+					self::set('has_exiftool', 1);
+					$has_exiftool = true;
+				}
+			} catch (Exception $e) {
 				self::set('has_exiftool', 0);
 				$has_exiftool = false;
-			} else {
-				self::set('has_exiftool', 1);
-				$has_exiftool = true;
+				Logs::warning(__METHOD__, __LINE__, 'exec is disabled, set has_exiftool to 0.');
```
Comment (d7415) [inline app/Configs.php]: Maybe `has_exiftool set to 0` or `has_exiftool has been set to 0` to clarify that this isn't an instruction.

File: app/Configs.php [hunk 2]
```diff
@@ -290,13 +296,19 @@ public static function hasFFmpeg()
...
+					self::set('has_ffmpeg', 1);
+					$has_ffmpeg = true;
+				}
+			} catch (Exception $e) {
 				self::set('has_ffmpeg', 0);
 				$has_ffmpeg = false;
-			} else {
-				self::set('has_ffmpeg', 1);
-				$has_ffmpeg = true;
+				Logs::warning(__METHOD__, __LINE__, 'exec is disabled, set_ffmpeg to 0.');
```
Comment (d7415) [inline app/Configs.php]: As above.

File: app/Http/Controllers/DiagnosticsController.php [hunk 3]
```diff
@@ -134,8 +134,22 @@ public function get_errors()
...
+			['mysql', 'mysqli'],
+			['mysql', 'pdo_mysql'],
+			['pgsql', 'pgsql'],
+			['pgsql', 'pdo_pgsql'],
+			['sqlite', 'sqlite3'],
+		];
+		try {
+			foreach ($db_possibilities as $db_possibility) {
+				if (DB::getDriverName() == $db_possibility[0] && !extension_loaded($db_possibility[1])) {
+					$errors[] = 'Error: PHP ' . $db_possibility[1] . ' extension not activated';
```
Comment (d7415) [inline app/Http/Controllers/DiagnosticsController.php]: For clarity, maybe this? Might need line wrapping.
```suggestion
					$errors[] = 'Error: ' . $db_possibility[0] . ' db driver selected and PHP ' . $db_possibility[1] . ' extension not activated';
```

**Review by d7415 — DISMISSED** [review]
Some wording suggestions, but the logic looks sound. Good thinking!

File: app/Http/Controllers/DiagnosticsController.php (hunk 3 above)
Comment (d7415) [reply app/Http/Controllers/DiagnosticsController.php]: I don't think I'd looked at these properly before. Maybe I should use them more often for the easy-commit, especially as we squash-merge.

**Pull request merged**